source/extensions/common/async_files/async_file_context_basic.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/extensions/common/async_files/async_file_context_impl.h"
+
+#include "absl/status/statusor.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace AsyncFiles {
+
+// The basic implementation of an AsyncFileContext - uses the manager thread pool and old-school
+// synchronous posix file operations.
+class AsyncFileContextBasic : public AsyncFileContextImpl {
```

Comment:
`AsyncFileContextImpl` seems to be an abstract class. Can we name it e.g. `AsyncFileContextBase` and make the actual implementation to be `AsyncFileContextBasicImpl`?

Reply:
Renamed the intermediate class to Base, and Basic to ThreadPool to make it clear that it's tied to a specific implementation.